### ISSUE TITLE
Cattle Hosts are not using External IP after upgrade to v1.3.3

### ISSUE THREAD

**Issue by clok** [rancher/rancher#7563]
**Rancher Versions:**
Server: v1.3.3
healthcheck: rancher/healthcheck:v0.1.0 (Stuck in `Initializing` for new hosts)
ipsec: rancher/net:v0.7.5
network-services: rancher/metadata:v0.6.8 & rancher/plugin-manager:v0.2.12
scheduler: rancher/scheduler:v0.4.0 (Which is reporting `Failed to acquire lock [SERVICE.1]`)
kubernetes (if applicable): N/A

**Docker Version:**
1.12.3

**OS and where are the hosts located? (cloud, bare metal, etc):**
Ubuntu 14.04.5 AWS

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher w/ external DB

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
Upgrade rancher server v1.2.0 to v1.3.3

**Results:** 
Cattle hosts that were previous working with External IP in AWS now report their Internal IP and are unable to allocate ports on external LoadBalancers.

Example would be a Rancher Agent reports the IP as `35.XX.XX.XX` before the upgrade then after the upgrade it reports `10.XX.XX.XX`.

This results in LoadBalancers reporting "Need to restart service reconcile" when performing an upgrade on the Rancher LB and then finally "Expected state running but got error: Timeout getting IP address".

Please note that the Rancher Scheduler is reporting `Failed to acquire lock [SERVICE.1]` and that the Healthcheck service is stuck `Initializing` with new Cattle hosts.

When adding a **NEW** Cattle host that has a Public IP assigned to it, the Cattle host agent reports back the Internal IP address. This means that new Cattle hosts are not function as well.

**Comment by clok** [comment]
We have many Rancher clusters that we manage and this was the first that we updated to `v1.3.3`. With `v1.3.2` we are not seeing this issue.

**Comment by clok** [comment]
Visual confirmation that new hosts do not have the "Use only private IP" checked on new host creation. These new hosts exhibit the behavior above.
<img width="466" alt="use-only-private-ip-not-checked" src="https://cloud.githubusercontent.com/assets/1429775/22335485/93cf889e-e3a4-11e6-9e6c-fdec3c47eda7.png">

**Comment by vincent99** [comment]
Your host registration URL is probably resolving to an internal IP.  The way it works (with no explicit CATTLE_AGENT_IP) isn't actually the agent reporting its IP, but the server looking at "What IP address did this request come in on".

**Comment by clok** [comment]
@vincent99 Thank you for your response. I did end up figuring out the root cause and resolving the issue. It was 2 fold.

1. User error. I did not read the patch notes clearly enough to recognize the bold message that was very important to the upgrade I performed. https://forums.rancher.com/t/rancher-release-v1-3-3/5303

```
When upgrading from prior v1.2.x releases, please check your infrastructure stacks to make sure your services are up to date. If an "Upgrade Available" button is shown, please upgrade to the latest service. Complete upgrading each individual infrastructure stack before moving on to the next stack.
```

2. In order to actually get the Infrastructure containers to upgrade AFTER migrating to v1.3.3 I needed to reboot the Cattle hosts. 

The agents still report the internal IP, which is actually okay, but the external IP is properly being used by the Rancher LB.

Thank you for your help and sorry for the false alarm.
